apis/datacatalog/v1alpha1/types.generated.go

```diff
@@ -12,8 +12,654 @@
+type CloudBigtableSystemSpec struct {
+	// Display name of the Instance. This is user specified and different from
+	//  the resource name.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.CloudBigtableSystemSpec.instance_display_name
+	InstanceDisplayName *string `json:"instanceDisplayName,omitempty"`
+}
+
+// +kcc:proto=google.cloud.datacatalog.v1.CloudSqlBigQueryConnectionSpec
+type CloudSQLBigQueryConnectionSpec struct {
+	// Cloud SQL instance ID in the format of `project:location:instance`.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.CloudSqlBigQueryConnectionSpec.instance_id
+	InstanceID *string `json:"instanceID,omitempty"`
+
+	// Database name.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.CloudSqlBigQueryConnectionSpec.database
+	Database *string `json:"database,omitempty"`
+
+	// Type of the Cloud SQL database.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.CloudSqlBigQueryConnectionSpec.type
+	Type *string `json:"type,omitempty"`
+}
+
+// +kcc:proto=google.cloud.datacatalog.v1.ColumnSchema
+type ColumnSchema struct {
+	// Required. Name of the column.
+	//
+	// Required. Name of the column.
+	//
+	//  Must be a UTF-8 string without dots (.).
+	//  The maximum size is 64 bytes.
+	// +required
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.column
+	Column *string `json:"column,omitempty"`
+
+	// Required. Type of the column.
+	//
+	//  Must be a UTF-8 string with the maximum size of 128 bytes.
+	// +required
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.type
+	Type *string `json:"type,omitempty"`
+
+	// Optional. Description of the column. Default value is an empty string.
+	//
+	//  The description must be a UTF-8 string with the maximum size of 2000
+	//  bytes.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.description
+	Description *string `json:"description,omitempty"`
+
+	// Optional. A column's mode indicates whether values in this column are
+	//  required, nullable, or repeated.
+	//
+	//  Only `NULLABLE`, `REQUIRED`, and `REPEATED` values are supported.
+	//  Default mode is `NULLABLE`.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.mode
+	Mode *string `json:"mode,omitempty"`
+
+	// Optional. Default value for the column.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.default_value
+	DefaultValue *string `json:"defaultValue,omitempty"`
+
+	// Optional. Ordinal position
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.ordinal_position
+	OrdinalPosition *int32 `json:"ordinalPosition,omitempty"`
+
+	// Optional. Most important inclusion of this column.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.highest_indexing_type
+	HighestIndexingType *string `json:"highestIndexingType,omitempty"`
+
+	// Optional. Schema of sub-columns. A column can have zero or more
+	//  sub-columns.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.ColumnSchema.subcolumns
+	Subcolumns []ColumnSchema `json:"subcolumns,omitempty"`
```

Comment:
This is a recursive field of the same struct. Is this tripping up CRD generation ?

Reply:
Marked it XPreserveUnknownFields true. 
Fixed CRD tooling to handle schemaless fields